### DISCUSSION TITLE
[Bug]: literal string "--" problem?

### DISCUSSION THREAD

**Discussion by dzzie** [sqlitebrowser/sqlitebrowser#3143] [General]
### What did you do?

Not sure where this bug lives, does seem like a bug though. 

no results: select 1 where "--" = "--"  

have to represent the literal string as either hex or with a concat

select 1  where x'2d2d' like "%-"||"-%"  

Sqllite queries through the ODBC driver were fine with the literal string "--"

(this is a boiled down test case, I was doing something like select * from main where file_path like "%--%" and it was failing
in sqllite browser but working over ODBC to same sqllite db)

### What did you expect to see?

seems like the following should give a result of 1 
        select 1 where "--" = "--" 
same as the following does:
        select 1 where "a" = "a"  

### What did you see instead?

no results returned

### DB4S Version

3.12.0

### What OS are you seeing the problem on?

Windows

### OS version

Windows 10

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by chrisjlocke** [comment]
In SQLite, literal strings are represented by single quotes.

    select *
    from table
    where field1='hello world'

Double-quotes are used for field names

    select *
    from table
    where "field1"='hello world'

![image](https://user-images.githubusercontent.com/7872218/185341217-6ce29b74-45aa-4f26-832f-98ba0cec82f9.png)

So in your example, replacing with single quotes returns one record.

![image](https://user-images.githubusercontent.com/7872218/185341543-ebbdfb57-a843-45e7-8e4d-613742528e75.png)

Bear in mind that this makes the apostrophe a 'control character' for SQLite, so it has to be escaped when doing string slicing like you're doing.  If you had a path, 'bob's nuts' in a variable PATH, you can't do a "select * from main where file_path like '%" + PATH + "%'" as the apostrophe in the PATH breaks the SQL.  You have to replace ' with '' (two single quotes).

So back to your example...

    select * from main where file_path like '%--%'

would work.
I guess the ODBC driver is switching that out for you behind the scenes?
Be interesting to throw that in the ODBC to see if it works, but that would definitely work in DB4S.

**Comment by chrisjlocke** [comment]
As this isn't an issue per se, moving this to the discussions.

**Comment by dzzie** [comment]
aghh ok thanks my bad

**Reply by chrisjlocke** [reply]
Not a problem at all

![image](https://user-images.githubusercontent.com/7872218/185406602-35a89c51-ff2f-486d-a895-4156162098c7.png)

We're here to help ... glad we could assist, and hope it actually worked your end. :)

**Comment by dzzie** [comment]
yup love the tool. literally use it everyday. thanks so much for all that you do :)
